ikarus/finiteelements/mechanics/nonlinearelastic.hh

```diff
@@ -152,12 +152,11 @@ public:
    */
   template <typename ScalarType, int strainDim, bool voigt = true>
   auto materialTangent(const Eigen::Vector<ScalarType, strainDim>& strain) const {
-    if constexpr (std::is_same_v<ScalarType, double>)
-      return mat_.template tangentModuli<strainType, voigt>(strain);
-    else {
+    if constexpr (Concepts::AutodiffScalar<ScalarType>) {
```

Comment:
Maybe we also want a private function getmaterial, which does this compile time if statement and returns the correct material object. Make sure that `decltype(auto)` is placed everywhere to not introduce copies